client-mr/src/main/java/org/apache/hadoop/mapreduce/task/reduce/RssBypassWriter.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.hadoop.mapreduce.task.reduce;
+
+import com.tencent.rss.common.exception.RssException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.io.IOUtils;
+import org.apache.hadoop.util.ReflectionUtils;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+
+public class RssBypassWriter {
```

Comment:
Why do we need write to disk?

Reply:
MR's merger will allocate to write very larger mapout to disk directly.

Reply:
Could we use merger's code?

Reply:
merger's code is difficult to use.

Reply:
Add some comments.

Reply:
Done

Reply:
I need the comment about explaining why we need this class, and we don't use this class usually.